### PR TITLE
rgw: Various cleanups and options update in rgw_admin.cc

### PR DESCRIPTION
**Pull request by joscollin**
Following cleanups and options updates are done in rgw-admin.cc. This is done as part of the doc update in https://github.com/ceph/ceph/pull/18267.

Changes:

1. rgw: update options in usage()

    Updated the display text of options in usage(). Dropped the repeated
    options --bucket and --pool and combined their text.

2. rgw: Fixed typo in options

    Fixed typo (underscore) in --show_log_entries, --show_log_sum and
    --skip_zero_entries.

3. rgw: Drop unused master_url and master_zonegroup

    Dropped the unused master_url and master_zonegroup. The value is set, but
    not used anywhere. As a consequence, dropped the options --master-url and
    --master-zonegroup too.

4. rgw: Drop unused options --rgw-realm, --rgw-zone and --rgw-zonegroup

    Dropped unused options --rgw-realm, --rgw-zone and --rgw-zonegroup. The
    realm_name, zone_name and zonegroup_name are not populated via this option
    respectively.

Signed-off-by: Jos Collin <jcollin@redhat.com>

### CODE REVIEW

**Comment by joscollin** [comment]
@oritwas Requesting your review. Please review this first, so that the doc PR https://github.com/ceph/ceph/pull/18267 can be updated appropriately. Thanks.

**Comment by joscollin** [comment]
@oritwas nudge :-)

File: src/rgw/rgw_admin.cc [hunk 1]
```diff
@@ -3321,7 +3310,7 @@ int main(int argc, const char **argv)
 	  return 1;
 	}
 	if (!realm_name.empty() && realm.get_name() != realm_name) {
-	  cerr << "mismatch between --rgw-realm " << realm_name << " and json input file name " <<
+	  cerr << "mismatch between rgw realm " << realm_name << " and json input file name " <<
```
Comment (oritwas) [inline src/rgw/rgw_admin.cc]: the realm is provided using --rgw-realm

File: src/rgw/rgw_admin.cc [hunk 2]
```diff
@@ -3730,7 +3719,7 @@ int main(int argc, const char **argv)
 
         if (zone_id.empty()) {
           if (zone_name.empty()) {
-            cerr << "no --zone-id or --rgw-zone name provided" << std::endl;
+            cerr << "no --zone-id or rgw zone name provided" << std::endl;
```
Comment (oritwas) [inline src/rgw/rgw_admin.cc]: the zone is provided using --rgw-zone

File: src/test/cli/radosgw-admin/help.t [hunk 3]
```diff
@@ -178,17 +178,12 @@
      --commit                  commit the period during 'period update'
      --staging                 get staging period info
      --master                  set as master
-     --master-url              master url
-     --master-zonegroup=<id>   master zonegroup id
      --master-zone=<id>        master zone id
-     --rgw-realm=<name>        realm name
```
Comment (oritwas) [inline src/test/cli/radosgw-admin/help.t]: needed

File: src/test/cli/radosgw-admin/help.t [hunk 4]
```diff
@@ -178,17 +178,12 @@
...
      --realm-id=<id>           realm id
      --realm-new-name=<name>   realm new name
-     --rgw-zonegroup=<name>    zonegroup name
```
Comment (oritwas) [inline src/test/cli/radosgw-admin/help.t]: same here

File: src/test/cli/radosgw-admin/help.t [hunk 5]
```diff
@@ -178,17 +178,12 @@
...
      --zonegroup-id=<id>       zonegroup id
      --zonegroup-new-name=<name>
                                zonegroup new name
-     --rgw-zone=<name>         name of zone in which radosgw is running
```
Comment (oritwas) [inline src/test/cli/radosgw-admin/help.t]: same here

**Comment by joscollin** [comment]
@oritwas Done. Please check now. Thanks.

**Comment by joscollin** [comment]
@oritwas Fixed the review comments. Could you please approve, if the changes are good ? Thanks.

File: src/rgw/rgw_admin.cc [hunk 6]
```diff
@@ -202,7 +202,7 @@ void usage()
   cout << "   --uid=<id>                user id\n";
   cout << "   --subuser=<name>          subuser name\n";
   cout << "   --access-key=<key>        S3 access key\n";
-  cout << "   --email=<email>\n";
+  cout << "   --email=<email>           The email address of the user\n";
```
Comment (oritwas) [inline src/rgw/rgw_admin.cc]: user's email address

File: src/rgw/rgw_admin.cc [hunk 7]
```diff
@@ -211,17 +211,17 @@ void usage()
   cout << "   --temp-url-key[-2]=<key>  temp url key\n";
   cout << "   --access=<access>         Set access permissions for sub-user, should be one\n";
   cout << "                             of read, write, readwrite, full\n";
-  cout << "   --display-name=<name>\n";
+  cout << "   --display-name=<name>     The display name of the user\n";
```
Comment (oritwas) [inline src/rgw/rgw_admin.cc]: user's display name

File: src/rgw/rgw_admin.cc [hunk 8]
```diff
@@ -211,17 +211,17 @@ void usage()
...
   cout << "   --admin                   set the admin flag on the user\n";
   cout << "   --system                  set the system flag on the user\n";
-  cout << "   --bucket=<bucket>\n";
-  cout << "   --pool=<pool>\n";
-  cout << "   --object=<object>\n";
-  cout << "   --date=<date>\n";
-  cout << "   --start-date=<date>\n";
-  cout << "   --end-date=<date>\n";
-  cout << "   --bucket-id=<bucket-id>\n";
+  cout << "   --bucket=<bucket>         Specify the bucket name. It is also used with quota command.\n";
```
Comment (oritwas) [inline src/rgw/rgw_admin.cc]: Also used by the quota command.

File: src/rgw/rgw_admin.cc [hunk 9]
```diff
@@ -211,17 +211,17 @@ void usage()
...
+  cout << "   --pool=<pool>             Specify the pool name. It can also be used to scan for leaked rados objects.\n";
```
Comment (oritwas) [inline src/rgw/rgw_admin.cc]: Also used to scan ...

File: src/rgw/rgw_admin.cc [hunk 10]
```diff
@@ -211,17 +211,17 @@ void usage()
...
+  cout << "   --object=<object>         Specify the object name\n";
```
Comment (oritwas) [inline src/rgw/rgw_admin.cc]: object name

File: src/rgw/rgw_admin.cc [hunk 11]
```diff
@@ -211,17 +211,17 @@ void usage()
...
+  cout << "   --date=<date>             The date needed for some commands\n";
```
Comment (oritwas) [inline src/rgw/rgw_admin.cc]: remove the some command and add the accepted date format

File: src/rgw/rgw_admin.cc [hunk 12]
```diff
@@ -211,17 +211,17 @@ void usage()
...
+  cout << "   --start-date=<date>       The start date needed for some commands\n";
```
Comment (oritwas) [inline src/rgw/rgw_admin.cc]: start date

File: src/rgw/rgw_admin.cc [hunk 13]
```diff
@@ -211,17 +211,17 @@ void usage()
...
+  cout << "   --end-date=<date>         The end date needed for some commands\n";
```
Comment (oritwas) [inline src/rgw/rgw_admin.cc]: end date

File: src/rgw/rgw_admin.cc [hunk 14]
```diff
@@ -211,17 +211,17 @@ void usage()
...
+  cout << "   --bucket-id=<bucket-id>   Specify the bucket id\n";
```
Comment (oritwas) [inline src/rgw/rgw_admin.cc]: bucket id

**Comment by joscollin** [comment]
@oritwas Fixed the review comments. Please check now. Thanks.

**Comment by joscollin** [comment]
@oritwas ping :-)

**Comment by joscollin** [comment]
@oritwas Fixed the review comments. Could you please verify ?

**Comment by joscollin** [comment]
[wip-jcollin-testing_11-19-2017](https://trello.com/c/bdkDTuGG/205-wip-jcollin-testing11-19-2017)

**Pull request merged**